app/models/submission.rb

```diff
@@ -96,6 +97,21 @@ def set_defaults
     self.content = Hash[STEPS.zip(STEPS.count.times.map {})] if content.blank?
   end
 
+  def apply_content_adjustments
+    return unless trial?
+    return unless content_changed?
+
+    step02_was = content_was[:step02] || {}
+    step02 = content[:step02] || {}
+
+    old_trait_descriptor_list = (step02_was[:trait_descriptor_list] || []).map(&:to_s).sort
+    new_trait_descriptor_list = (step02[:trait_descriptor_list] || []).map(&:to_s).sort
```

Comment:
Please compare without sorting. Due to problems with possible name duplicates, I had to change the semantics of trait descriptor list and now the order of contents is meaningful. Therefore, simply changing places of same two traits in the list should also delete step03 content.